src/goto-programs/goto_program.h

```diff
@@ -541,6 +541,10 @@ class goto_programt
     /// Number unique per function to identify loops
     unsigned loop_number = 0;
 
+    /// We record the original loop number for some instrumended instructions
+    /// so that we know which loop they are transformed from.
+    optionalt<unsigned> original_loop_number;
+
```

Comment:
Please drop, this is not needed, and costs extra memory.

Reply:
Sure. I plan to instead define `ID_original_loop_number` and store the original loop number in the field `ID_original_loop_number` of the lhs `exprt` of instrumented assignments such as loop-havoc. Does it sound OK to you?

Reply:
No, please use a map.

Reply:
Thank you, I will store the map from instrumented instructions to their original loop number in `code_contractst`. And the functions I used to decided whether an instruction is a loop-havoc or instrumented instruction will be no longer needed with such map.